src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -249,15 +249,23 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
     } else {
       tracksToStop = mediaStreamToRelease.getTracks();
     }
-    for (const track of tracksToStop) {
-      this.logger.info(`stopping ${track.kind} track`);
-      track.stop();
-    }
     for (const kind in this.activeDevices) {
       if (this.activeDevices[kind] && this.activeDevices[kind].stream === mediaStreamToRelease) {
         this.activeDevices[kind] = null;
+        if (
+          kind === 'video' &&
+          this.boundAudioVideoController &&
+          this.boundAudioVideoController.videoTileController.hasStartedLocalVideoTile()
+        ) {
+          this.boundAudioVideoController.videoTileController.stopLocalVideoTile();
+        }
       }
     }
+
+    for (const track of tracksToStop) {
```

Comment:
Not a fan of code comments, but in this case, I think it's worth explaining why we need to set the active device `null` first and then stop tracks.

Reply:
Added comment